### ISSUE TITLE
Skip using a callback for `default.qubit.jax` and `diff_method="parameter-shift"`

### ISSUE THREAD

**Issue by antalszava** [PennyLaneAI/pennylane#3259]
### Feature details

Specifying `diff_method="parameter-shift"` and transforming the QNode with `jax.jit` results in applying the JAX JIT interface. When used with the `"default.qubit.jax"` device, computing the gradient involves using `jax.pure_callback`. Using `jax.pure_callback`, however, is not required with this device because it is meant to natively used with JAX data structures.

### Implementation

With the current architecture of PennyLane, the simplest change would involve update the JAX JIT interface to have logic for the `default.qubit.jax` device.

### How important would you say this feature is?

1: Not important. Would be nice to have.

**Comment by astralcai** [comment]
![](https://media.tenor.com/TigFWS1gf50AAAAM/2000anos-depois.gif)

`default.qubit.jax` does not exist anymore

**Comment by josh146** [comment]
Does the same issue still occur in the new implementation of JAX default.qubit?

**Comment by albi3ro** [comment]
But this is still a problem that we still haven't really fixed. Right now, we only send interface data to the device when `diff_method="backprop"`.  Otherwise, we assume the device requires pure numpy.

This was something that came up when porting to the new device interface. Who is responsible for converting the parameters to numpy.  The device? Or the core pennylane workflow? So our current fix is always converting things to numpy unless we want to do backprop.

But we could definitely use a better solution. Especially if we want to jit more of the simulation, or if we want to be able to keep data on the gpu end-to-end.

**Comment by astralcai** [comment]
The original premise of the issue was that we had a device that "is meant to natively used with JAX data structures", which we don't anymore? Maybe we can rephrase the issue to something that makes sense in the current context?

**Comment by albi3ro** [comment]
We have a device that can be natively used with all interface data structures.
